source/lib/cdk-textract-stack.ts

```diff
@@ -122,6 +127,276 @@ export class CdkTextractStack extends cdk.Stack {
       }
     );
 
+    // Layers for lambda functions
+    const cicdBotoLoc = lambda.Code.fromBucket(
+      s3.Bucket.fromBucketName(this, "solutionBucketBoto", "SOURCE_BUCKET"),
+      "SOLUTION_NAME/CODE_VERSION/boto3-layer.zip"
+    );
+
+    const cicdPDFLoc = lambda.Code.fromBucket(
+      s3.Bucket.fromBucketName(this, "solutionBucketPDF", "SOURCE_BUCKET"),
+      "SOLUTION_NAME/CODE_VERSION/searchable-pdf-1.0.jar"
+    );
+
+    // If a local yarn deploy is used, the two lambdas draw their code from a local directory.
+
+    const yarnBotoLoc = lambda.Code.fromAsset("lambda/boto3/boto3-layer.zip");
+
+    const yarnPDFLoc = lambda.Code.fromAsset("lambda/pdfgenerator");
+
+    const helperLayer = new lambda.LayerVersion(
+      this,
+      this.resourceName("HelperLayer"),
+      {
+        code: lambda.Code.fromAsset("lambda/helper"),
+        compatibleRuntimes: [lambda.Runtime.PYTHON_3_7],
+        license: "Apache-2.0"
+      }
+    );
+
+    const textractorLayer = new lambda.LayerVersion(
+      this,
+      this.resourceName("Textractor"),
+      {
+        code: lambda.Code.fromAsset("lambda/textractor"),
+        compatibleRuntimes: [lambda.Runtime.PYTHON_3_7],
+        license: "Apache-2.0"
+      }
+    );
+
+    const boto3Layer = new lambda.LayerVersion(
+      this,
+      this.resourceName("Boto3"),
+      {
+        code: props.isCICDDeploy ? cicdBotoLoc : yarnBotoLoc,
+        compatibleRuntimes: [lambda.Runtime.PYTHON_3_7],
+        license: "Apache-2.0"
+      }
+    );
+
+    const elasticSearchLayer = new lambda.LayerVersion(
+      this,
+      this.resourceName("ElasticSearchLayer"),
+      {
+        code: lambda.Code.fromAsset("lambda/elasticsearch/es.zip"),
+        compatibleRuntimes: [lambda.Runtime.PYTHON_3_7],
+        license: "Apache-2.0"
+      }
+    );
+
+    // global 
+    let createKendraIndexCustomResource = null;
+    let kendraRoleArn = null;
+    if(props.enableKendra){
```

Comment:
I would like to avoid such a massive if condition in the same code.
Can we move this to a separate .ts file.

This type of snippets are a bit diffcult to read especially in such a huge file.

How about changing this to 

     if(props.enableKendra){ 
      indexId,rolearn = createandGetKendraRelatedResources()
      updataLambdaWithKendraInformation (kendra_information, lambda_list)
}